todoapp/app.py

```diff
@@ -64,13 +64,14 @@ def format_datetime(value):
     # flask-login
     login_manager = LoginManager()
     login_manager.init_app(app)
-    login_manager.login_view = "login"
+
     client = WebApplicationClient(CLIENT_ID)
-    csrfState = ''.join(random.choice(string.ascii_letters) for _ in range(10))
+    csrf_state = ''
 
     @login_manager.unauthorized_handler
     def unauthenticated():
-        request_uri = client.prepare_request_uri('https://github.com/login/oauth/authorize', state=csrfState)
+        csrf_state = ''.join(random.choice(string.ascii_letters) for _ in range(10))
```

Comment:
Worth noting that this will only work for a small number of users, otherwise they'll interrupt each others authentication flow. If using the state to prevent CSRF, we usually store the relevant state somewhere on the client, e.g. set it in their session, and then we can determine the correct state per user - two authenticating users will then not get in each others way. [See this doc](https://auth0.com/docs/protocols/state-parameters) for more info